File: data/code_pairs/cpp/1691.cpp
#include <iostream>

namespace c_funloc_tests_5 {
    void sub1();
    int func0(int desired_retval);

    void sub0() {
        void (*my_c_funptr)() = nullptr;
        int (*my_c_funcptr)(int) = nullptr;

        my_c_funptr = &sub1;
        my_c_funptr(); // Calls sub1

        my_c_funcptr = &func0;
        std::cout << "Result of func0: " << my_c_funcptr(42) << std::endl;
    }

    void sub1() {
        std::cout << "sub1 called" << std::endl;
    }

    int func0(int desired_retval) {
        return desired_retval;
    }
} // namespace c_funloc_tests_5

File: data/code_pairs/fortran/1691.f90
module c_funloc_tests_5
  use, intrinsic :: iso_c_binding, only: c_funloc, c_funptr

contains

  subroutine sub0() bind(c)
    type(c_funptr) :: my_c_funptr

    my_c_funptr = c_funloc(sub1) 
    call sub1()

    my_c_funptr = c_funloc(func0)
    print *, "Result of func0: ", func0(42)
  end subroutine sub0

  subroutine sub1() 
    print *, "sub1 called"
  end subroutine sub1

  function func0(desired_retval) result(res)
    use, intrinsic :: iso_c_binding, only: c_int
    integer(c_int), value :: desired_retval
    integer(c_int) :: res
    res = desired_retval
  end function func0
end module c_funloc_tests_5